dev-ruby/fiber-local/fiber-local-1.0.0.ebuild

```diff
@@ -0,0 +1,25 @@
+# Copyright 2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+USE_RUBY="ruby26 ruby27 ruby30 ruby31"
+RUBY_FAKEGEM_RECIPE_TEST="rspec3"
+RUBY_FAKEGEM_EXTRADOC="README.md"
+RUBY_FAKEGEM_GEMSPEC="${PN}.gemspec"
+inherit ruby-fakegem
+
+DESCRIPTION="Provides a class-level mixin to make fiber local state easy"
+HOMEPAGE="https://github.com/socketry/fiber-local"
+SRC_URI="https://github.com/socketry/fiber-local/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="$(ver_cut 1)/$(ver_cut 1-2)"
```

Comment:
I'm not sure what you mean here? Can you explain the reasoning in more detail? What you you mean with `^fiber-local-1.0.0`? I'm not sure what the carrot does in this example because it is not ruby or gentoo syntax for dependencies. I'm also not sure what you would expect to happen with the subslot.

Note that they are not harmless. They bring another thing to take into account and they clutter up overviews of packages so they make maintenance more difficult. I want to understand the benefits before accepting those costs.

Reply:
Bleh sorry was working on other languages, I meant `gem 'fiber-local', '~> 1.0.0`.  I did this specifically based on our IRC conversation:

```
2022-06-19 01:56:38     @graaff matoro: I see now that you also change slots. That can't be done without a slotmove, not sure if that is worth the effort.
2022-06-19 01:57:36     matoro  don't have a choice, new packages specify pinned versions
2022-06-19 01:58:37     @graaff I understand that for e.g. a dependency on ~> foo-3.0 it is more communicative to write foo:3 than foo:0 (if the slot was never set) but the latter works fine.
2022-06-19 02:00:53     matoro  for the slots, what to do if we have a foo-3.0 and foo-4.0 in tree and the package needs ~> foo-3.0
2022-06-19 02:01:04     matoro  and both foo-3.0 and foo-4.0 have SLOT=0
2022-06-19 02:01:35     @graaff matoro: in that case we can either do a slotmove or use something like =foo-3*
2022-06-19 02:02:43     @graaff But I don't think there are many packages left where this happens. In the past I tried to be "smart" and avoid slots when the changelog seemed to suggest there were no breaking changes but in the end that appoach just caused more problems.
2022-06-19 02:02:56     matoro  ok one more, this one was the reason I started the SLOT stuff originally - what about if the package specifies ~> foo-3.6 and wee have 3.11 in tree
2022-06-19 02:03:12     matoro  so what that needs is >=foo-3.6:3
2022-06-19 02:03:29     matoro  can't just do >=foo-3.6 because that will match foo-4.x
2022-06-19 02:04:31     @graaff =foo-3* >=foo-3.6
2022-06-19 02:04:51     @graaff But for packages where this actually happens we should probably consider fixing the slot.
```

So because of this conversation, I made sure that all the new ebuilds I added have a SLOT, so that they won't require a slotmove in the future were something to specify a restrictive dependency.

Reply:
My question is about the subslot, not the slot itself.

Reply:
We don't use subslots anywhere in dev-ruby. Can you explain how you think this helps with specific situations that may arise with dev-ruby packages.

The http gem depends on  llhttp-ffi ~> 0.4.0, so the correct way to deal with that is to use the "0.4" slot here and refer to that in the http ebuild. I don't see how a subslot is going to help here.

Reply:
There was an instance in the original PR (I don't recall which, looks like it's gone now) where there were two reverse dependencies with different strictness levels.  One depended on `~> 1.0.0` and the other on `~> 1.0`.

Reply:
Yes, that happens sometimes because semantic versioning is not enforced in rubygems. In this case there are two options: a) (preferred) use the "1.0" slot for both packages and if newer minor versions are released (e.g. 1.1.0) make the package with ~> 1.0 depend on both versions with "|| ( )", or b) tweak the gemspec of the package depending on ~> 1.0.0 to be more lenient. 

It is not clear to me how this relates to the subslot or what purpose a subslot would have.

Reply:
With a subslot aren't we able to make the first package depend on `dev-ruby/example:1/1.0` and the second package depend on `dev-ruby/example:1`?  This seems much cleaner to me than having to change reverse dependencies on a new release or change what the original author specified.

Reply:
That approach has the same drawback as using a dependency like "=dev-ruby/example-1.0*". It is hard to track this when working on the package itself, and consequently easy to break the dependency tree when cleaning up old versions in slots. I'd prefer to have that explicit by using slots only.

Reply:
Isn't that what pkgcheck is for?  To yell about dropping versions that are still required by something?

So which is preferred:
* Set `SLOT=1`, package A depends on `dev-ruby/example:1`, package B depends on `=dev-ruby/example-1.0*`
* or, set `SLOT=1.0`, package A depends on `=dev-ruby/example-1*`, package B depends on `dev-ruby/example:1.0`

Reply:
@graaff ping on this

Reply:
I see examples of both in tree. Which do you want matoro to pursue, @graaff?